src/main/scala/beam/agentsim/agents/parking/ChoosesParking.scala

```
@@ -171,7 +172,9 @@ trait ChoosesParking extends {
     val lastLeg = vehicleTrip.last.beamLeg
     val activityType = nextActivity(data).get.getType
     val remainingTripData = calculateRemainingTripData(data)
-    val parkingDuration = nextActivity(data).map(_.getEndTime - lastLeg.endTime).getOrElse(0.0)
+    val parkingDuration =
+      nextActivity(data).map(_.getEndTime.orElse(beam.UNDEFINED_TIME) - lastLeg.endTime).getOrElse(0.0)
```

This is how the old code would work - but maybe it is worth re-thinking. If endtime is empty then we could yield a 0.0 also? Or maybe this is a scenario where we SHOULD throw an exception?

---

I think it should be 0.0 because the last Activity is usually a Home activity with undefined end time.